app/controllers/concerns/api/v2/parameter_deprecation.rb

```diff
@@ -0,0 +1,61 @@
+# This module adds a pre-action filter that will handle deprecated properties.
```

Comment:
Why not just use something like https://github.com/theforeman/foreman/blob/develop/app/controllers/api/v2/template_combinations_controller.rb#L81 instead of this whole complicated module? API deprecations should be a very rare event, as the API is considered stable, and should only be done when there is no other option.